pkg/reconciler/cache/replication/replication_reconcile_unstructured.go

```diff
@@ -0,0 +1,225 @@
+)
+
+// reconcileUnstructuredObject makes sure that the given unstructuredCachedObject of the given GVR under the given key from the local shard is replicated to the cache server.
+//
+// this method handles the following cases:
+//  1. creation of the object in the cache server
+//     happens when the unstructuredCacheObject is nil
+//  2. deletion of the object from the cache server
+//     happens when either of the following is true:
+//     - the unstructuredLocalObject object is nil
+//     - the unstructuredLocalObject was deleted
+//  3. modification of the object to match the original/local object
+//     happens when either of the following is true:
+//      - the unstructuredLocalObject's metadata doesn't match the unstructuredCacheObject
+//        as of today we compare only the annotations and the labels
+//      - the unstructuredLocalObject's spec doesn't match the unstructuredCacheObject
+//      - the unstructuredLocalObject's status doesn't match the unstructuredCacheObject
+func (c *controller) reconcileUnstructuredObject(ctx context.Context, cluster logicalcluster.Name, gvr *schema.GroupVersionResource, unstructuredCacheObject *unstructured.Unstructured, unstructuredLocalObject *unstructured.Unstructured) error {
+	if unstructuredLocalObject == nil {
+		return c.handleUnstructuredObjectDeletion(ctx, cluster, gvr, unstructuredCacheObject, unstructuredLocalObject)
+	}
+	if unstructuredLocalObject.GetDeletionTimestamp() != nil {
+		return c.handleUnstructuredObjectDeletion(ctx, cluster, gvr, unstructuredCacheObject, unstructuredLocalObject)
+	}
+
+	if unstructuredCacheObject == nil {
+		// TODO: in the future the original RV will have to be stored in an annotation (?)
+		// so that the clients that need to modify the original/local object can do it
+		unstructuredLocalObject.SetResourceVersion("")
+		_, err := c.dynamicCacheClient.Cluster(cluster).Resource(*gvr).Namespace(unstructuredLocalObject.GetNamespace()).Create(ctx, unstructuredLocalObject, metav1.CreateOptions{})
+		return err
+	}
+
+	metaChanged, err := ensureUnstructuredMeta(unstructuredCacheObject, unstructuredLocalObject)
+	if err != nil {
+		return err
+	}
+	specChanged, err := ensureUnstructuredSpec(unstructuredCacheObject, unstructuredLocalObject)
+	if err != nil {
+		return err
+	}
+	statusChanged, err := ensureUnstructuredStatus(unstructuredCacheObject, unstructuredLocalObject)
+	if err != nil {
+		return err
+	}
+	if !specChanged && !metaChanged && !statusChanged {
+		return nil
+	}
+
+	if specChanged || metaChanged {
+		_, err := c.dynamicCacheClient.Cluster(cluster).Resource(*gvr).Namespace(unstructuredCacheObject.GetNamespace()).Update(ctx, unstructuredCacheObject, metav1.UpdateOptions{})
+		return err
+	}
+	if !statusChanged {
+		return nil
+	}
+	_, err = c.dynamicCacheClient.Cluster(cluster).Resource(*gvr).Namespace(unstructuredCacheObject.GetNamespace()).UpdateStatus(ctx, unstructuredCacheObject, metav1.UpdateOptions{})
+	return err
+}
```

Comment:
these subresource distinctions surprise me. Why not disable the status subresource in the cache? Then we can skip all of this dance.

Reply:
I can look into that but I think it comes from the extension server we use. Might require yet another patch in the fork.

Reply:
@sttts you were right, turning off subresources did the trick :) PTAL - https://github.com/kcp-dev/kcp/pull/2063

Reply:
🥳